src/box/box.cc

```diff
@@ -796,14 +797,25 @@ box_check_replication(void)
 static int
 box_check_listen(void)
 {
-	int count = cfg_getarr_size("listen");
-	for (int i = 0; i < count; i++) {
-		const char *source = cfg_getarr_elem("listen", i);
-		assert(source != NULL);
-		if (box_check_uri(source, "listen") != 0)
```

Comment:
You should patch `box_check_replication` as well in this patch, and drop `box_check_uri`.

Reply:
Fixed